### DISCUSSION TITLE
Query string in Catalog List Items returning Invalid Control Character response

### DISCUSSION THREAD

**Discussion by brickbuilderbob** [saleweaver/python-amazon-sp-api#71] [Q&A] [answered]
Hi,

I'm trying to use the Catalog().list_items endpoint, and am getting the following error when my query string includes blank spaces:

"json.decoder.JSONDecodeError: Invalid control character at: line 4 column 197 (char 218)"

I've been experimenting with various query strings, and have found that including blank spaces is causing the error. If I use comma separated keywords, I can get a response. I also can get a response if I join keywords with a "+" character. Unfortunately, neither of these approaches returns the result I'm expecting when I search Seller Central's Catalog manually. Below are the examples of query strings I've tried.

Query Strings searching for lego set number 10270

Search String 
"lego 10270" - this returns the error message above and includes a blank space separating "lego" and "10270"
"lego,10270" - returns a result, but includes too many results from other products
"lego+10270" - returns completely inaccurate results

More simply stated, I would like to use a search query string that replicates the same results I'm seeing in Seller Central when I search the catalog.

Below is my code:

from sp_api.api import Catalog
from sp_api.base import SellingApiException, Marketplaces
import json
import ast

print('Start API')

QueryString = str('lego 10270')
QueryContext = str('Toys')
Marketplace = str('ATVPDKIKX0DER')

print(QueryString)

Response = Catalog().list_items(MarketplaceId = Marketplace, Query = QueryString, QueryContextId = QueryContext)

print(Response.payload)

**Comment by saleweaver** [comment, marked as answer]
Thank you for bringing this up. The endpoint is now fixed to allow spaces in Query. 

The wrapper doesn't alter the response in any way and there's nothing I can do about the other questions raised.

**Reply by brickbuilderbob** [reply]
Thanks! I installed the latest release and the list_items endpoint is now accepting my queries with spaces in them.

**Comment by zonwizard** [comment]
Hi, 
we are facing the same issue when there is a space in the Sku

Marketplace = str('ATVPDKIKX0DER')
sku = 'HP envy_6230_'
Response = Catalog().list_items(MarketplaceId = Marketplace, SellerSKU=sku)

the request url is encoded with a + instead of the %20 in place of the space:
 'https://sellingpartnerapi-eu.amazon.com/catalog/v0/items?MarketplaceId=APJ6JRA9NG5V4&SellerSKU=HP+envy_6230_'

Thanks a lot for your help

Guido

**Reply by saleweaver** [reply]
Hi, can you please check out from master and try if that works?

**Reply by zonwizard** [reply]
Yes, it worked.
Do you know when this will be available in the official relased version?

Thanks a lot

**Reply by saleweaver** [reply]
I'll release now

**Reply by saleweaver** [reply]
https://github.com/saleweaver/python-amazon-sp-api/releases/tag/v0.13.6

**Comment by zonwizard** [comment]
You are the best!
